features/admin.rules.v1/components/rule-conditions.tsx

```diff
@@ -140,344 +162,391 @@ const RuleConditions: FunctionComponent<RulesComponentPropsInterface> = ({
+                            endAdornment: (
+                                <>
+                                    { isInitialLoading || isFiltering || isResourceDetailsLoading ? (
+                                        <CircularProgress color="inherit" size={ 20 } />
+                                    ) : null }
+                                    { params.InputProps.endAdornment }
+                                </>
+                            )
+                        } }
+                    />
+                ) }
+                renderOption={ (props: any, option: { label: string; id: string, isDisabled: boolean }) => {
+                    if (option.id === "more-items") {
                         return (
-                            <li { ...props }>
+                            <li
+                                { ...props }
+                                // TODO: Check the issue why className is not working.
+                                // And move styles to a css file.
+                                style={ {
+                                    color: "#666666",
+                                    fontStyle: "italic",
+                                    padding: "12px 15px 5px",
+                                    pointerEvents: option.isDisabled ? "none" : "auto"
+                                } }
+                                key={ option.id }
+                            >
                                 { option.label }
                             </li>
                         );
-                    } }
-                />
-            );
-        };
+                    }
 
-        /**
-         * Resource list select component.
-         *
-         * @returns Resource list select component.
-         */
-        const ResourceListSelect: FunctionComponent<ResourceListSelectProps> = (props: any) => {
-            const { initialResourcesLoadUrl, filterBaseResourcesUrl } = props;
+                    return (
+                        <li { ...props }>
+                            { option.label }
+                        </li>
+                    );
+                } }
+            />
+        );
+    };
 
-            const valueReferenceAttribute: string = findMetaValuesAgainst?.value?.valueReferenceAttribute || "id";
-            const valueDisplayAttribute: string = findMetaValuesAgainst?.value?.valueDisplayAttribute || "name";
+    /**
+     * Resource list select component.
+     *
+     * @returns Resource list select component.
+     */
+    const ResourceListSelect: FunctionComponent<ResourceListSelectProps> = (props: ResourceListSelectProps) => {
+        const {
+            ruleId,
+            conditionId,
+            setIsResourceMissing,
+            expressionValue,
+            expressionId,
+            expressionField,
+            findMetaValuesAgainst,
+            initialResourcesLoadUrl,
+            filterBaseResourcesUrl
```

Comment:
shall we include a default value for component Id here and use it in the Select component?

Reply:
Added.